### PR TITLE
added across common migration methods

### PR DESCRIPTION
**Pull request by swalker125**
initial commit for fpl-data-migration-tool. 
Moved across common methods for migration. Repo will need to be twisted to apply required migrations.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+def versions = [
+        ccdStoreClient: '4.4.1',
+        commonsLang3: '3.7',
+        commonsIo: '2.5',
+        feignHttpClient: '9.5.1',
+        gradlePitest: '1.3.0',
+        jsonAssert: '1.2.3',
+        junit: '4.12',
+        lombok: '1.16.16',
+        pitest: '1.3.2',
```
Comment (geek-soft) [inline build.gradle]: Lots of needed reps, like mutation testing framework for example. Other dips would be good to update as well.

Comment (geek-soft) [reply build.gradle]: So much rubbish, I guess due to effort required we can clean that later.

File: src/main/java/uk.gov.hmcts.reform.fpl.ccddatamigration/CcdDataMigrationApplication.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.hmcts.reform.fpl.ccddatamigration;
+
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import uk.gov.hmcts.reform.authorisation.healthcheck.ServiceAuthHealthIndicator;
+
+@EnableFeignClients(basePackages = {"uk.gov.hmcts.reform.finrem"})
```
Comment (geek-soft) [inline src/main/java/uk.gov.hmcts.reform.fpl.ccddatamigration/CcdDataMigrationApplication.java]: Remove any reference to `finrem`.

File: src/test/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# Enable default health indicators
+# Enable disk space health check
+management.endpoint.health.show-details="ALWAYS"
+management.endpoint.health.cache.time-to-live=4000
+management.endpoints.web.base-path=/
+management.endpoints.web.path-mapping.health=health
+management.endpoints.web.exposure.include=info,health,metrics
+
+logging.level.org.springframework.web=ERROR
+logging.level.uk.gov.hmcts.reform.finrem.ccddatamigration-=DEBUG
```
Comment (geek-soft) [inline src/test/resources/application.properties]: Make sure all reference to `uk.gov.hmcts.reform.finrem` are gone.

File: src/test/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,54 @@
+info.app.name=${spring.application.name}
```
Comment (geek-soft) [inline src/test/resources/application.properties]: Are these test values different from defaults in YAML file? Is that really needed? Maybe default would do for testing purpose.

Comment (swalker125) [reply src/test/resources/application.properties]: currently no yaml, do you mean our one from fpl-ccd-configuration

**Pull request merged**